camp/accounts/views.py

```diff
@@ -9,3 +16,34 @@ class ProfileDetailView(LoginRequiredMixin, DetailView):
 
     def get_object(self):
         return self.request.user
+
+
+# Put a create page under /accounts/profile/create, and similar for edit.
+
+
+class MembershipListView(ListView):
+    model = Membership
+    template_name = "account/membership_list.html"
+    context_object_name = "memberships"
+
+
+class MembershipDetailView(DetailView):
+    model = Membership
+
+
+class MembershipCreateView(CreateView):
+    model = Membership
+    fields = ["joined", "nickname", "game", "user"]
```

Comment:
When joining a game, the only field from this list we should allow is "nickname".

We don't want to let the user specified when they joined. Django will set it automatically.
We don't need the user to specify the game. We should read this from the request.
We shouldn't let the user pick what user they're creating a profile for. It should always be for the current user.

See `camp.game.views.CreateRulesetView.form_valid` for an example of automatically setting a newly created object's `game` relation based on the request.

Also, if you find it easier to use function-based views, that's perfectly fine too.